### PR TITLE
[CHK-3162] Update eventstore deadletter 

### PR DESCRIPTION
**Pull request by EmanueleBVtech**
#### List of Changes
- added new classes to manipulate data for transactionInfo model
- implemented logic to aggregate and enhance data stored in dead letter
- added logic to extract transactionId from event binary data
- Added unit test to match code coverage requirements
- now targeting commons 1.25.0

#### Motivation and Context
We need to enhance the information in the dead letter document. To do so we want to obtain a set of informations about the transaction starting from the transactionId wich is found in the event data. This data is currently already stored with the dead letter as the only relevant data.

#### How Has This Been Tested?
Unit test

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/217

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -289,26 +302,23 @@
...
                     <execution>
                         <id>compile</id>
+                        <phase>compile</phase>
                         <goals>
                             <goal>compile</goal>
                         </goals>
                         <configuration>
                             <sourceDirs>
-                                <sourceDir>${project.basedir}/src/main/kotlin</sourceDir>
+                                <source>src/main/kotlin</source>
```
Comment (infantesimone) [inline pom.xml:311]: Why is this change necessary?

File: src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/configurations/QueuesConsumerConfig.kt [hunk 2]
```diff
@@ -46,4 +50,13 @@ class QueuesConsumerConfig {
     ): StorageQueueMessageSource {
         return StorageQueueMessageSource(queueNameClosureEvents, storageQueueTemplate)
     }
+
+    @Bean
+    fun strictSerializerProviderV2(): StrictJsonSerializerProvider =
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/configurations/QueuesConsumerConfig.kt:55]: why v2?

File: pom.xml (hunk 1 above)
Comment (EmanueleBVtech) [reply pom.xml:311]: It isn't necessary, this was probably an automated repair from IntelliJ.
This sometimes happens because IntelliJ has some issues with WSL and sometimes it breaks the configurations,
but i can revert the line to the previous configuration.

Do you prefer a rollback to the previous configuration?

File: src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/configurations/QueuesConsumerConfig.kt (hunk 2 above)
Comment (EmanueleBVtech) [reply src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/configurations/QueuesConsumerConfig.kt:55]: We chose the v2 implementation because we wanted to be more up-to-date and considered v1 as obsolete.

File: pom.xml (hunk 1 above)
Comment (infantesimone) [reply pom.xml:311]: if not needed, rollback

**Comment by infantesimone** [comment]
@EmanueleBVtech this PR is blocked by https://github.com/pagopa/pagopa-ecommerce-commons/pull/217?

File: src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/common.kt [hunk 3]
```diff
@@ -19,9 +24,24 @@ fun writeEventToDeadLetterCollection(
     payload: ByteArray,
     queueName: String,
     checkPointer: Checkpointer,
-    deadLetterEventRepository: DeadLetterEventRepository
+    deadLetterEventRepository: DeadLetterEventRepository,
+    transactionInfoBuilder: TransactionInfoBuilder,
+    strictSerializerProviderV2: StrictJsonSerializerProvider
 ): Mono<Unit> {
+    val transactionId =
+        BinaryData.fromBytes(payload)
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/common.kt:32]: what do you think about using `toObjectAsync` instead of `toObject`? So, something like:

`toObjectAsync().map().onErrorResume()`

and in `onErrorResume` add some log

File: pom.xml [hunk 4]
```diff
@@ -273,6 +285,7 @@
                         <plugin>spring</plugin>
                         <plugin>lombok</plugin>
                     </compilerPlugins>
+                    <jvmTarget>1.8</jvmTarget>
```
Comment (infantesimone) [inline pom.xml:288]: 👎

File: src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/common.kt (hunk 3 above)
Comment (EmanueleBVtech) [reply src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/common.kt:32]: In case we fail to read the transactionId, do we still want to proceed and store the deadletter with all the other info (basically transactionInfo will be null), or should the process be stopped and the error logged?

File: pom.xml (hunk 4 above)
Comment (EmanueleBVtech) [reply pom.xml]: Done

File: src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/common.kt (hunk 3 above)
Comment (infantesimone) [reply src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/common.kt:32]: store in deadletter in any cas

File: src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/common.kt [hunk 5]
```diff
@@ -19,9 +24,24 @@ fun writeEventToDeadLetterCollection(
...
+            )
+            .event
+            .transactionId
+
     val eventData = payload.toString(StandardCharsets.UTF_8)
+
+    // recover here info on event based on transactionId
+    val transactionInfo = transactionInfoBuilder.getTransactionInfoByTransactionId(transactionId)
+
     CommonLogger.logger.debug("Read event from queue: {}", eventData)
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/common.kt:45]: let's review the order of operations, first log and retrieve the entire event and then enrich it? let's keep the flow consistent with the logs (maybe add a log for data enrichment too)

File: src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/TransactionInfoBuilder.kt [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+import it.pagopa.ecommerce.commons.generated.server.model.TransactionStatusDto
+import it.pagopa.ecommerce.commons.utils.v2.TransactionUtils.getTransactionFee
+import it.pagopa.ecommerce.transactions.scheduler.repositories.TransactionsEventStoreRepository
+import java.util.*
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+import reactor.core.publisher.Mono
+
+@Service
+class TransactionInfoBuilder(
```
Comment (infantesimone) [inline src/main/kotlin/it/pagopa/ecommerce/transactions/scheduler/deadletter/TransactionInfoBuilder.kt:20]: This class doesn't seem to reflect the Java builder pattern, why is it called builder?
